Lib/test/test_concurrent_futures.py

```diff
@@ -234,14 +253,20 @@ def _prime_executor(self):
 
     @contextlib.contextmanager
     def _assert_logged(self, msg):
-        if self.executor_type is futures.ProcessPoolExecutor:
-            # No easy way to catch the child processes' stderr
+        if self.log_queue is not None:
             yield
+            output = []
+            while True:
+                try:
+                    output.append(self.log_queue.get_nowait().getMessage())
+                except queue.Empty:
```

Comment:
You could catch an exception outside of the loop.

Reply:
Oh, you're right.